course_discovery/apps/edx_haystack_extensions/distinct_counts/backends.py

```
@@ -0,0 +1,331 @@
+        # Therefore, it should always be configured with an aggregation_key.
+        if not self._aggregation_key:
+            raise RuntimeError('DistinctCountsSearchQuery cannot run without aggregation_key.')
+
+        final_query = self.build_query()
+        search_kwargs = self.build_params(spelling_query=spelling_query)
+
+        if kwargs:
+            search_kwargs.update(kwargs)
+
+        # Use the DistinctCountsElasticsearchBackendWrapper to execute the query so that distinct hit and query
+        # counts may be computed.
+        backend = DistinctCountsElasticsearchBackendWrapper(self.backend, self._aggregation_key)
+        results = backend.search(final_query, **search_kwargs)
+
+        self._results = results.get('results', [])
+        self._hit_count = results.get('hits', 0)
+        self._distinct_hit_count = results.get('distinct_hits', 0)
+        self._facet_counts = self.post_process_facets(results)
+        self._spelling_suggestion = results.get('spelling_suggestion', None)
+
+    def run_mlt(self, **kwargs):
+        """
+        Run a more-like-this query.
+
+        The DistinctCountsSearchQuery class is only meant to be used when distinct hit and facet counts are needed. The
+        logic necessary for computing distinct counts has not been implemented for more-like-this queries. Therefore,
+        this method will raise an error if called.
+        """
+        raise NotImplemented('DistinctCountsSearchQuery.run_mlt is not supported.')
+
+    def run_raw(self, **kwargs):
+        """
+        Run a raw search query.
+
+        The DistinctCountsSearchQuery class is only meant to be used when distinct hit and facet counts are needed. The
+        logic necessary for computing distinct counts has not been implemented for raw search queries. Therefore, this
+        method will raise an error if called.
+        """
+        raise NotImplemented('DistinctCountsSearchQuery.run_raw is not supported.')
```

It's necessary because I don't want people to be able to use this Query object in an unexpected way. I'd prefer it to fail loudly if/when someone uses it like this so that that person sees the issue right away.